### PR TITLE
Extend printer to handle non-empty domain

### PR DESCRIPTION
**Pull request by gramalingam**
Signed-off-by: Ganesan Ramalingam <grama@microsoft.com>

**Description**

The printer does not currently handle a non-empty domain (used for ops in domains other than the standard one).
Extend the printer to handle it.

### CODE REVIEW

File: onnx/defs/printer.cc [hunk 1]
```diff
@@ -319,7 +319,10 @@ void ProtoPrinter::print(const AttrList& attrlist) {
 void ProtoPrinter::print(const NodeProto& node) {
   output_ << std::setw(indent_level) << ' ';
   printSet("", ", ", "", node.output());
-  output_ << " = " << node.op_type();
+  output_ << " = ";
+  if (node.domain() != "")
+    output_ << node.domain() << ".";
```
Comment (daquexian) [inline onnx/defs/printer.cc:324]: "ai.onnx" and "" both mean the default domain. Should ops with domain "ai.onnx" have consistent behavior with those with "" domain?

Comment (gramalingam) [reply onnx/defs/printer.cc:324]: yes, that's right. (Does it have any implication to this PR?)

Comment (daquexian) [reply onnx/defs/printer.cc:324]: > Does it have any implication to this PR?

Yes, maybe the line 323 should be changed to `if (node.domain() != "" && node.domain() != "ai.onnx")`

Comment (gramalingam) [reply onnx/defs/printer.cc:324]: I think it is better in the current form. Both forms are legal and mean the same thing. The parser/printer are just convenience utilities to convert the proto format to/from a readable textual form. It is useful to preserve the distinction between the two different, but equivalent, forms in the parser/printer.

**Review by linkerzhang — APPROVED** [review]
LGTM.

**Review by wschin — APPROVED** [review]

**Pull request merged**